File: buffer/cpu/src/jvmMain/java/com/wsr/cpu/index/JIndex.java
package com.wsr.cpu.index;

import java.nio.ByteBuffer;

public class JIndex {
    public native void gather(ByteBuffer x, ByteBuffer y, int i, int j, int k, ByteBuffer result);
    public native void scatterAdd(ByteBuffer x, ByteBuffer y, int i, int j, int k, int b, ByteBuffer result);
}

File: buffer/cpu/src/jvmAndroidMain/kotlin/com/wsr/knist/cpu/index/JIndex.kt
package com.wsr.knist.cpu.index


object JIndex {
    external fun gather(x: Long, y: Long, i: Int, j: Int, k: Int, result: Long)
    external fun scatterAdd(x: Long, y: Long, i: Int, j: Int, k: Int, b: Int, result: Long)
}
